src/twitter.ts

```diff
@@ -0,0 +1,154 @@
+import axios from "axios";
+import { OpenSeaCollection } from "opensea-js/lib/types";
+import { config } from "./config";
+import { openSeaPort } from "./opensea";
+import { PartyEvent } from "./types";
+import {
+  bestUserName,
+  getIsNewPartyWithContribution,
+  // getIsPartyHalfWay,
+} from "./utils";
+
+/**
+ * queries opensea api for the nft's opensea colleciton
+ * @param {PartyEvent} event
+ * @returns {Promise<OpenSeaCollection>}
+ */
+const getOpenseaCollection = async (
+  event: PartyEvent
+): Promise<OpenSeaCollection> => {
+  const {
+    party: { nftContractAddress, nftTokenId },
+  } = event;
+  try {
+    const r = await axios.get(
+      `https://api.opensea.io/api/v1/asset/${nftContractAddress}/${nftTokenId}`,
+      {
+        headers: {
+          "X-API-KEY": config.openSeaApiKey,
+        },
+      }
+    );
+    if (r) {
+      return r.data.collection;
+    } else {
+      return undefined;
+    }
+  } catch (e) {
+    console.error(e);
+    return undefined;
+  }
+};
+
+/**
+ * extracts the twitter username from the opensea collection if available. otherwise
+ *   extracts the opensea collection name
+ * @param {OpenSeaCollection} collection - opensea collection
+ * @returns {string} - twitter username or collection name
+ */
+const getTwitterHandleOrNameFromCollection = (
+  collection: OpenSeaCollection
+): string => {
+  if (!collection) {
+    return "";
+  }
+
+  // twitter_username is returned by the API, but is not on the OpenSeaCollection type
+  // @ts-ignore
+  const twitterUsername = collection.twitter_username;
+  return twitterUsername
+    ? // add "@" to the collection twitter username so that twitter will hyperlink
+      // to that collection's twitter account
+      `@${twitterUsername}`
+    : collection.name;
+};
+
+const getTwitterHandleOrNameFromEvent = async (
+  event: PartyEvent
+): Promise<string> => {
+  if (
+    event.eventType === "contribution" ||
+    event.eventType === "finalization"
+  ) {
+    const collection = await getOpenseaCollection(event);
+    const twitterHandleOrName =
+      getTwitterHandleOrNameFromCollection(collection);
+    return twitterHandleOrName;
+  } else {
+    return "";
+  }
+};
+
+const swapText = async (event: PartyEvent): Promise<string> => {
```

Comment:
nit but why is this named `swapText`? seems like `eventText` would be a more natural name

Reply:
copy paste from the discord file. i agree, i'll change it to `eventText`